### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by windhamwong**
Huge update for Android setup.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,38 +39,72 @@ I've created [an example project](https://github.com/evollu/react-native-fcm/tre
...
+
+### Edit `android/app/src/main/AndroidManifest.xml`:
+
+**NOTE:** The resource `@mipmap/ic_launcher` is referring to `android/app/src/res/mipmap-<resolution>` folder. Feel free to change the filename if you have a customised icon for notification.
 
 ```diff
   <application
     ...
     android:theme="@style/AppTheme">
++    <meta-data android:name="com.google.firebase.messaging.default_notification_icon" android:resource="@mipmap/ic_launcher" />
```
Comment (evollu) [inline README.md]: is it dup of 109-110?

File: README.md [hunk 2]
```diff
@@ -39,38 +39,72 @@ I've created [an example project](https://github.com/evollu/react-native-fcm/tre
...
+
+**NOTE:** Please refer to https://firebase.google.com/docs/android/setup
+
+```
+dependencies {
+    // ...
+    compile project(':react-native-fcm')
+
+    // ...
+    implementation 'com.google.firebase:firebase-core:16.0.1'
```
Comment (evollu) [inline README.md]: where does this come from? why does it have different version between core and messaging?

File: README.md (hunk 1 above)
Comment (evollu) [reply README.md]: android Lolipop+ requires a white icon, so usually it will be different from ic_launcher icon

File: README.md (hunk 2 above)
Comment (evollu) [reply README.md]: we support version using variables. I think that is the way other packages doing it as well. Can we go with that approach in the readme?
```
def DEFAULT_FIREBASE_CORE_VERSION           = "12.+"
def DEFAULT_FIREBASE_MESSAGING_VERSION      = "12.+"
```

Comment (windhamwong) [reply README.md]: Just reverify a bit these lines, realised that should modify the src for not using these DEFAULT version instead. The "12.+" limits the version to 12.x.x which is too old for now. I will commit a new one again

File: android/build.gradle [hunk 3]
```diff
@@ -3,9 +3,9 @@ apply plugin: 'com.android.library'
 def DEFAULT_COMPILE_SDK_VERSION             = 27
 def DEFAULT_BUILD_TOOLS_VERSION             = "27.0.3"
 def DEFAULT_TARGET_SDK_VERSION              = 27
-def DEFAULT_GOOGLE_PLAY_SERVICES_VERSION    = "12.+"
-def DEFAULT_FIREBASE_CORE_VERSION           = "12.+"
-def DEFAULT_FIREBASE_MESSAGING_VERSION      = "12.+"
+def DEFAULT_GOOGLE_PLAY_SERVICES_VERSION    = "+"
+def DEFAULT_FIREBASE_CORE_VERSION           = "+"
+def DEFAULT_FIREBASE_MESSAGING_VERSION      = "+"
```
Comment (windhamwong) [inline android/build.gradle]: This change is to select the latest version instead of limiting to 12 only.

**Pull request merged**